flink-runtime/src/main/java/org/apache/flink/runtime/state/KeyedStateBackend.java

```diff
@@ -69,25 +69,26 @@
 	 * @param stateDescriptor The identifier for the state. This contains name and can create a default state value.
 
 	 * @param <N> The type of the namespace.
+	 * @param <V> The type of the values in the state.
 	 * @param <S> The type of the state.
 	 *
 	 * @return A new key/value state backed by this backend.
 	 *
 	 * @throws Exception Exceptions may occur during initialization of the state and should be forwarded.
 	 */
 	@SuppressWarnings({"rawtypes", "unchecked"})
-	<N, S extends State> S getPartitionedState(
+	<N, V, S extends State<V>> S getPartitionedState(
```

Comment:
I think we don't need the additional `V` parameter. It's never used and `State<?>` works just as while while `State<V>` just pretends to add more type safety. Because both ways work I would prefer to go with the solution that doesn't add additional generic parameters to methods.

Same holds for `mergePartitionedStates()`.